spire/templates/shared-app-redis.yml

```diff
@@ -0,0 +1,107 @@
+        - FromPort: 6379
+          IpProtocol: tcp
+          SourceSecurityGroupId: !Ref SourceSecurityGroupId1
+          ToPort: 6379
+      Tags:
+        - { Key: Name, Value: !Sub "${RootStackName}_shared_app_redis" }
+        - { Key: prx:meta:tagging-version, Value: "2021-04-07" }
+        - { Key: prx:cloudformation:stack-name, Value: !Ref AWS::StackName }
+        - { Key: prx:cloudformation:stack-id, Value: !Ref AWS::StackId }
+        - { Key: prx:cloudformation:root-stack-name, Value: !Ref RootStackName }
+        - { Key: prx:cloudformation:root-stack-id, Value: !Ref RootStackId }
+        - { Key: prx:ops:environment, Value: !Ref EnvironmentType }
+        - { Key: prx:dev:application, Value: Common }
+      VpcId: !Ref VpcId
+
+  SharedAppRedisSubnetGroup:
+    Type: AWS::ElastiCache::SubnetGroup
+    Condition: IsPrimaryRegion
+    Properties:
+      Description: !Sub Shared ${EnvironmentType} App Redis subnet group
+      SubnetIds:
+        - !Ref VpcPrivateSubnet1Id
+        - !Ref VpcPrivateSubnet2Id
+        - !Ref VpcPrivateSubnet3Id
+      Tags:
+        - { Key: prx:meta:tagging-version, Value: "2021-04-07" }
+        - { Key: prx:cloudformation:stack-name, Value: !Ref AWS::StackName }
+        - { Key: prx:cloudformation:stack-id, Value: !Ref AWS::StackId }
+        - { Key: prx:cloudformation:root-stack-name, Value: !Ref RootStackName }
+        - { Key: prx:cloudformation:root-stack-id, Value: !Ref RootStackId }
+        - { Key: prx:ops:environment, Value: !Ref EnvironmentType }
+        - { Key: prx:dev:application, Value: Common }
+
+  SharedAppRedisCluster:
+    Type: AWS::ElastiCache::CacheCluster
+    DeletionPolicy: Delete
+    UpdateReplacePolicy: Delete
+    Condition: IsPrimaryRegion
+    Properties:
+      AutoMinorVersionUpgrade: true
+      CacheNodeType: !If
+        - IsProduction
+        - !Ref ProductionInstanceType
+        - !Ref StagingInstanceType
+      CacheSubnetGroupName: !Ref SharedAppRedisSubnetGroup
+      Engine: redis
+      EngineVersion: 6.x
+      NumCacheNodes: 1
+      PreferredMaintenanceWindow: sun:09:00-sun:10:00
+      Tags:
+        - { Key: prx:meta:tagging-version, Value: "2021-04-07" }
+        - { Key: prx:cloudformation:stack-name, Value: !Ref AWS::StackName }
+        - { Key: prx:cloudformation:stack-id, Value: !Ref AWS::StackId }
+        - { Key: prx:cloudformation:root-stack-name, Value: !Ref RootStackName }
+        - { Key: prx:cloudformation:root-stack-id, Value: !Ref RootStackId }
+        - { Key: prx:ops:environment, Value: !Ref EnvironmentType }
+        - { Key: prx:dev:application, Value: Common }
+      VpcSecurityGroupIds:
+        - !Ref SharedAppRedisSecurityGroup
+
+Outputs:
+  CacheName:
+    Value: !If [IsPrimaryRegion, !Ref SharedAppRedisCluster, !Ref "AWS::NoValue"]
+  CacheEndpointAddress:
+    Description: Cache endpoint hostname
+    Value: !If [IsPrimaryRegion, !GetAtt SharedAppRedisCluster.RedisEndpoint.Address, !Ref "AWS::NoValue"]
```

Comment:
This cache only exists in the primary region - so hopefully this NoValue logic works to pass through templates?

The apps that use this ENV should handle it not being set, or being unavailable.